pkg/integrations/prometheus/promql/parser.go

```diff
@@ -0,0 +1,564 @@
+	timeRange := mQueryReqs[0].TimeRange
+
+	arithmeticOperation := structs.QueryArithmetic{}
+	var lhsRequest, rhsRequest []*structs.MetricsQueryRequest
+	var lhsQueryArth, rhsQueryArth []*structs.QueryArithmetic
+
+	if constant, ok := expr.LHS.(*parser.NumberLiteral); ok {
+		arithmeticOperation.ConstantOp = true
+		arithmeticOperation.Constant = constant.Val
+	} else {
+		lhsRequest, _, lhsQueryArth, err = parsePromQLQuery(expr.LHS.String(), timeRange.StartEpochSec, timeRange.EndEpochSec, myid)
+		if err != nil {
+			return mQueryReqs, queryArithmetic, err
+		}
+		arithmeticOperation.LHS = lhsRequest[0].MetricsQuery.HashedMName
+		queryArithmetic = append(queryArithmetic, lhsQueryArth...)
+	}
+
+	if constant, ok := expr.RHS.(*parser.NumberLiteral); ok {
+		arithmeticOperation.ConstantOp = true
+		arithmeticOperation.Constant = constant.Val
+	} else {
+		rhsRequest, _, rhsQueryArth, err = parsePromQLQuery(expr.RHS.String(), timeRange.StartEpochSec, timeRange.EndEpochSec, myid)
+		if err != nil {
+			return mQueryReqs, queryArithmetic, err
+		}
+		arithmeticOperation.RHS = rhsRequest[0].MetricsQuery.HashedMName
+		queryArithmetic = append(queryArithmetic, rhsQueryArth...)
+	}
+	arithmeticOperation.Operation = getArithmeticOperation(expr.Op)
+	queryArithmetic = append(queryArithmetic, &arithmeticOperation)
+
+	if mQueryReqs[0].MetricsQuery.MQueryAggs == nil {
+		mQueryReqs = lhsRequest
+	} else if mQueryReqs[0].MetricsQuery.HashedMName == uint64(0) {
+		// This means there is a common group by on multiple metrics separated by operators.
+		// So, we need to append the aggregations to each of the Request in LHS and RHS.
+		// And we can discard the current aggregation in the mQueryReqs[0]
+		rhsRequest = appendMetricAggsToTheMQuery(rhsRequest, &mQueryReqs[0].MetricsQuery)
+		mQueryReqs = appendMetricAggsToTheMQuery(lhsRequest, &mQueryReqs[0].MetricsQuery)
+	} else {
+		mQueryReqs = append(mQueryReqs, lhsRequest...)
+	}
+	mQueryReqs = append(mQueryReqs, rhsRequest...)
+
+	return mQueryReqs, queryArithmetic, nil
+}
+
+func appendMetricAggsToTheMQuery(mQueryReqs []*structs.MetricsQueryRequest, mQuery *structs.MetricsQuery) []*structs.MetricsQueryRequest {
+	for _, mQueryReq := range mQueryReqs {
+		currentAggs := mQueryReq.MetricsQuery.MQueryAggs
+		for currentAggs.Next != nil {
+			currentAggs = currentAggs.Next
+		}
+		currentAggs.Next = mQuery.MQueryAggs
+
+		mQueryReq.MetricsQuery.Groupby = mQuery.Groupby
+		mQueryReq.MetricsQuery.SelectAllSeries = mQuery.SelectAllSeries
+		mQueryReq.MetricsQuery.TagsFilters = mQuery.TagsFilters
+	}
+	return mQueryReqs
+}
+
+func HandleNumberLiteral(literal *parser.NumberLiteral, mQuery *structs.MetricsQuery) {
+	// Check if the ValueList is nil, if so, initialize it
+	if mQuery.Function.ValueList == nil {
+		mQuery.Function = structs.Function{
+			ValueList: []string{},
+		}
+	}
+	// Append the number literal to the ValueList
+	mQuery.Function.ValueList = append(mQuery.Function.ValueList, literal.String())
+}
```

Comment:
No it does not need to be public, but I thought I would need this function, 
but as of now we are not using this...
But I kept it because for the queries such as `1 + 1` or `vector(1) + vector(1)` something like this... 
I wanted to deal with these queries when I am actually working on Formulas. So for now I wanted to leave it like that.

But I can remove this for this PR and add the function in the Formulas function. So shall I remove it now?

Reply:
I see. Probably remove it here if it won't be used in this PR, but it's not a big deal either way

Reply:
I guess, I will remove it. Makes it more clean.